crates/subspace-networking/src/node_runner.rs

```diff
@@ -463,29 +479,54 @@ where
                     }
                 }
             }
-            SwarmEvent::OutgoingConnectionError { peer_id, error } => match error {
-                DialError::Transport(ref addresses) => {
-                    for (addr, _) in addresses {
-                        debug!(?error, ?peer_id, %addr, "SwarmEvent::OutgoingConnectionError (DialError::Transport) for peer.");
-                        if let Some(peer_id) = peer_id {
-                            self.networking_parameters_registry
-                                .remove_known_peer_addresses(peer_id, vec![addr.clone()])
-                                .await;
+            SwarmEvent::OutgoingConnectionError { peer_id, error } => {
+                if let Some(peer_id) = &peer_id {
+                    // Create or extend temporary ban, but only if we are not offline
+                    if let Some(shared) = self.shared_weak.upgrade() {
+                        // One peer is possibly a node peer is connected to, hence expecting more
+                        // than one for online status
+                        if shared.connected_peers_count.load(Ordering::Relaxed) > 1 {
+                            let should_temporary_ban = match &error {
+                                DialError::Transport(addresses) => {
+                                    // Ignoring other errors, those are likely temporary ban errors
+                                    !matches!(
+                                        addresses.first(),
```

Comment:
Why did you match the first address only?

Reply:
They typically have the same error. Basically libp2p will try to dial all the addresses known for a peer and if the first one (typically publicly reachable) is unavailable and we get here, then others are unavailable too and likely for the same reason. So I thought checking the first one is sufficient to check for transport error. They certainly all failed, we just don't check why exactly beyond the most likely (first) address that is potentially reachable.